server/channel_recommendations.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/iomodo/mattermost-plugin-suggestions/server/ml"
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/model"
+)
+
+func mapToSlice(m map[string]*model.Channel) []*model.Channel {
+	channels := make([]*model.Channel, len(m))
+	index := 0
+	for _, channel := range m {
+		channels[index] = channel
+		index++
+	}
+	return channels
+}
+
+func (p *Plugin) isChannelOk(channelID string) bool {
+	posts, err := p.API.GetPostsForChannel(channelID, 0, 1)
+
+	if err != nil || len(posts.Order) == 0 {
+		return false
+	}
+	return true
+}
+
+func (p *Plugin) getChannelListFromRecommendations(recommendations []*recommendedChannel) []*model.Channel {
+	sort.Slice(recommendations, func(i, j int) bool {
+		return recommendations[i].Score > recommendations[j].Score
+	})
+	channels := make([]*model.Channel, 0)
+	for _, rec := range recommendations {
+		channel, err := p.API.GetChannel(rec.ChannelID)
+		if err != nil {
+			mlog.Error(fmt.Sprintf("Can't get channel - %v, err is %v", rec.ChannelID, err.Error()))
```

Comment:
We actually recommend using fields like `mlog.String("channel_id", rec.ChannelID)` and `` instead of logging a formatted string since that gives the logger some nicer json output. For example, this would ideally look like
```suggestion
			mlog.Error("Can't get channel", mlog.String("channel_id", rec.ChannelID), mlog.Err(err))
```